tests/end_to_end/gcc-division/mwcc-o4p-out.c

```diff
@@ -305,16 +303,16 @@ void test_u32_div(u32 u) {
     func_00400090(sp8 / 100000000);
     func_00400090(sp8 >> 0x1EU);
     func_00400090((u32) MULTU_HI(-3, sp8) >> 0x1EU);
-    temp_r3_10 = MULTU_HI(5, sp8);
+    temp_r3_10 = sp8 / 858993459;
     func_00400090((u32) (((u32) (sp8 - temp_r3_10) >> 1U) + temp_r3_10) >> 0x1EU);
-    temp_r3_11 = MULTU_HI(3, sp8);
+    temp_r3_11 = sp8 / 1431655765;
     func_00400090((u32) (((u32) (sp8 - temp_r3_11) >> 1U) + temp_r3_11) >> 0x1EU);
-    func_00400090(MULTU_HI(2, sp8));
+    func_00400090(sp8 / 2147483648);
     temp_r3_12 = MULTU_HI(-3, sp8);
     func_00400090((u32) (((u32) (sp8 - temp_r3_12) >> 1U) + temp_r3_12) >> 0x1FU);
-    temp_r3_13 = MULTU_HI(3, sp8);
+    temp_r3_13 = sp8 / 1431655765;
     func_00400090((u32) (((u32) (sp8 - temp_r3_13) >> 1U) + temp_r3_13) >> 0x1FU);
-    func_00400090(sp8 / 4294967294);
+    func_00400090((u32) (sp8 / 2) >> 0x1FU);
 }
 
 void test_u32_mod(u32 u) {
```

Comment:
I just noticed that this function tests the exact same as the div one, seems it was copy-pasted while forgetting to change / -> %.

Reply:
I pushed a fix for this. Wasn't able to update gcc-o2.s though since I've lost track of which compiler that corresponded to.